### PR TITLE
Ports VG's false briefcase

### PR DESCRIPTION
**Pull request by AzuleUtama**
**What does this PR do:**
This ports across the false-bottomed briefcase from VG, so full credit for the item goes to them.

The case can be bought from the traitor uplink and will accept any gun that's normal sized or smaller. You need to click at least 1 tile away from you to fire the gun, clicking directly next to you will simply smack them with the case instead. Firing the briefcase will retain most characteristics from the gun itself, including suppression, spread, ammo type and recoil. You can open the false bottom with a screwdriver to remove or reload the gun inside.

The case comes empty rather than starting with an SMG loaded like VG's. This is due to coders/admins stating they did not want more firearms added to traitor loadouts. Instead I've lowered the price to 3TC, which is cheap enough for people to load it with any gun they want from the uplink.

The briefcase itself can be examined to reveal it's true nature, and anyone can screwdriver it open to check for guns inside.

**Changelog:**
:cl:
add: Adds false-bottomed briefcase from VG.
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
Is this basically the briefcase gun but you can put any normal-sized gun into it to fire with? I think I rather just have the briefcase gun as this sounds a little weird

**Comment by AzuleUtama** [comment]
> Is this basically the briefcase gun but you can put any normal-sized gun into it to fire with? I think I rather just have the briefcase gun as this sounds a little weird

It's VG's briefcase gun except it comes with the compact sniper rifle loaded rather than an SMG. The ability to replace the gun inside is also in their code.

**Comment by Spartan6** [comment]
This was sounding okay until you added the Compact Sniper and started messing with it.

Replace the compact sniper with a revolver and leave the sniper rifle alone, as that should be a PR of its own.

**Comment by AzuleUtama** [comment]
> This was sounding okay until you added the Compact Sniper and started messing with it.
> 
> Replace the compact sniper with a revolver and leave the sniper rifle alone, as that should be a PR of its own.

Fair point, my original intention was to find a way to make the compact sniper more desirable but the briefcase is something that would be better as a separate item it seems.

I think having it include the revolver would make the item too expensive for people to consider buying, bearing in mind it would need to cost more than 13TC. Another option I was thinking of when porting was having it be empty to start with and letting people load whatever gun/item they wanted into the false bottom., be it a revolver, stechkin, taser etc. Would that be a better way to implement it?

**Comment by datlo** [comment]
I dont really see the point of having it start with a gun loaded. Would be more fun to have it empty with a low price tag, yeah.

**Comment by TDSSS** [comment]
Compact sniper is very underused, I like adding it to this.

**Comment by taukausanake** [comment]
I'd keep it with the SMG or give it a handgun. The better option would be to make it start empty. Make it cheaper to reflect that.

> Compact sniper is very underused

That gun shouldn't have even been added to start with. The argument before it was added was there were too many guns and more would add bloat. The rifle itself was outclassed by other guns and its disuse as seen is pretty good proof to that.
Can we also agree that a rifle, even a compact one, sounds ridiculous when placed inside a briefcase?

**Comment by AzuleUtama** [comment]
Following the initial feedback, I've reverted the changes to the rifle and made the briefcase start off empty at 3TC. Let me know if this a better option / cheap enough price.

**Comment by lolzyking** [comment]
Could probably add this to the agent bundle...

**Comment by Krysonism** [comment]
> The briefcase itself can be examined to reveal it's true nature

Validhunters will just taze / disarm all lawyers carrying a briefcase to check for this.

**Comment by lolzyking** [comment]
We already have the Dr bees brief case, and if valid hunters start rando checking all brief cases.....well that could be fixed with a good ol Bwoink about how only sec is allowed to random search, and only on certain alert levels.

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 1]
```diff
@@ -20,10 +20,78 @@
...
+
+/obj/item/storage/briefcase/false_bottomed/afterattack(atom/A, mob/user, flag, params)
+	..()
+	if(stored_item && istype(stored_item, /obj/item/gun) && get_dist(A, user) > 1)
+		var/obj/item/gun/stored_gun = stored_item
+		stored_gun.afterattack(A, user, flag, params)
+	return
+
+/obj/item/storage/briefcase/false_bottomed/attackby(var/obj/item/item, mob/user)
+	if(istype(item, /obj/item/screwdriver))
```
Comment (Couls) [inline code/game/objects/items/weapons/storage/briefcase.dm]: isscrewdriver(item)

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 2]
```diff
@@ -20,10 +20,78 @@
...
+		if(!bottom_open && !busy_hunting)
+			to_chat(user, "You begin to hunt around the rim of \the [src]...")
+			busy_hunting = 1
```
Comment (Couls) [inline code/game/objects/items/weapons/storage/briefcase.dm]: TRUE

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 3]
```diff
@@ -20,10 +20,78 @@
...
+	desc = "It's made of AUTHENTIC faux-leather and has a price-tag still attached. This one feels a bit heavier than normal for how much fits in it."
+	icon_state = "briefcase"
+	force = 8.0
+	throw_speed = 1
+	throw_range = 3
+	w_class = WEIGHT_CLASS_BULKY
+	max_w_class = WEIGHT_CLASS_SMALL
+	max_combined_w_class = 10
+
+	var/busy_hunting = 0
```
Comment (Couls) [inline code/game/objects/items/weapons/storage/briefcase.dm]: FALSE

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 4]
```diff
@@ -20,10 +20,78 @@
...
+	var/bottom_open = 0 //is the false bottom open?
```
Comment (Couls) [inline code/game/objects/items/weapons/storage/briefcase.dm]: FALSE

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 5]
```diff
@@ -20,10 +20,78 @@
...
+/obj/item/storage/briefcase/false_bottomed/attackby(var/obj/item/item, mob/user)
+	if(istype(item, /obj/item/screwdriver))
+		if(!bottom_open && !busy_hunting)
+			to_chat(user, "You begin to hunt around the rim of \the [src]...")
+			busy_hunting = 1
+			if(do_after(user, 20, target = src))
+				if(user)
+					to_chat(user, "You pry open the false bottom!")
+				bottom_open = 1
+			busy_hunting = 0
```
Comment (Couls) [inline code/game/objects/items/weapons/storage/briefcase.dm]: FALSE

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 6]
```diff
@@ -20,10 +20,78 @@
...
 	new /obj/item/ammo_box/magazine/sniper_rounds/soporific(src)
-	new /obj/item/suppressor/specialoffer(src)
\ No newline at end of file
+	new /obj/item/suppressor/specialoffer(src)
+
+/obj/item/storage/briefcase/false_bottomed
+	name = "briefcase"
+	desc = "It's made of AUTHENTIC faux-leather and has a price-tag still attached. This one feels a bit heavier than normal for how much fits in it."
+	icon_state = "briefcase"
+	force = 8.0
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: no `.0` please.

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 7]
```diff
@@ -20,10 +20,78 @@
...
 	new /obj/item/clothing/under/syndicate/sniper(src)
 	new /obj/item/ammo_box/magazine/sniper_rounds/soporific(src)
 	new /obj/item/ammo_box/magazine/sniper_rounds/soporific(src)
-	new /obj/item/suppressor/specialoffer(src)
\ No newline at end of file
+	new /obj/item/suppressor/specialoffer(src)
+
+/obj/item/storage/briefcase/false_bottomed
+	name = "briefcase"
+	desc = "It's made of AUTHENTIC faux-leather and has a price-tag still attached. This one feels a bit heavier than normal for how much fits in it."
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: I'd rather the description be identical to the regular briefcase so people can't meta it PURELY by looking at it.

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 8]
```diff
@@ -20,10 +20,78 @@
...
+
+	var/busy_hunting = FALSE
+	var/bottom_open = FALSE //is the false bottom open?
+	var/obj/item/stored_item = null //what's in the false bottom. If it's a gun, we can fire it
+
+/obj/item/storage/briefcase/false_bottomed/Destroy()
+	if(stored_item)//since the stored_item isn't in the briefcase' contents we gotta remind the game to delete it here.
+		qdel(stored_item)
+		stored_item = null
+	..()
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: `return ..()`

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 9]
```diff
@@ -20,10 +20,78 @@
...
+	max_combined_w_class = 10
+
+	var/busy_hunting = FALSE
+	var/bottom_open = FALSE //is the false bottom open?
+	var/obj/item/stored_item = null //what's in the false bottom. If it's a gun, we can fire it
+
+/obj/item/storage/briefcase/false_bottomed/Destroy()
+	if(stored_item)//since the stored_item isn't in the briefcase' contents we gotta remind the game to delete it here.
+		qdel(stored_item)
+		stored_item = null
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: replace all of this with `QDEL_NULL(stored_item)`

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 10]
```diff
@@ -20,10 +20,78 @@
...
+	..()
+
+/obj/item/storage/briefcase/false_bottomed/afterattack(atom/A, mob/user, flag, params)
+	..()
+	if(stored_item && istype(stored_item, /obj/item/gun) && get_dist(A, user) > 1)
+		var/obj/item/gun/stored_gun = stored_item
+		stored_gun.afterattack(A, user, flag, params)
+	return
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: no trailing returns....

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 11]
```diff
@@ -20,10 +20,78 @@
...
+
+/obj/item/storage/briefcase/false_bottomed/Destroy()
+	if(stored_item)//since the stored_item isn't in the briefcase' contents we gotta remind the game to delete it here.
+		qdel(stored_item)
+		stored_item = null
+	..()
+
+/obj/item/storage/briefcase/false_bottomed/afterattack(atom/A, mob/user, flag, params)
+	..()
+	if(stored_item && istype(stored_item, /obj/item/gun) && get_dist(A, user) > 1)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: do a !adjacent check

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 12]
```diff
@@ -20,10 +20,78 @@
...
+		var/obj/item/gun/stored_gun = stored_item
+		stored_gun.afterattack(A, user, flag, params)
+	return
+
+/obj/item/storage/briefcase/false_bottomed/attackby(var/obj/item/item, mob/user)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: implied var, also `item/item` is bad representation; make it `I`.

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 13]
```diff
@@ -20,10 +20,78 @@
...
+	if(isscrewdriver(item))
+		if(!bottom_open && !busy_hunting)
+			to_chat(user, "You begin to hunt around the rim of \the [src]...")
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: no `\the`

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 14]
```diff
@@ -20,10 +20,78 @@
...
+			bottom_open = FALSE
+	else if(bottom_open)
+		if(stored_item)
+			to_chat(user, "<span class='warning'>There's already something in the false bottom!</span>")
+			return
+		if(item.w_class > WEIGHT_CLASS_NORMAL)
+			to_chat(user, "<span class='warning'>\The [item] is too big to fit in the false bottom!</span>")
+			return
+		if(!user.drop_item(item))
+			user << "<span class='warning'>\The [item] is stuck to your hands!</span>"
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: Again.

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 15]
```diff
@@ -20,10 +20,78 @@
...
+			return
+
+		stored_item = item
+		max_w_class = WEIGHT_CLASS_NORMAL - stored_item.w_class
+		item.forceMove(null) //null space here we go - to stop it showing up in the briefcase
+		to_chat(user, "You place \the [item] into the false bottom of the briefcase.")
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: again.

File: code/game/objects/items/weapons/storage/briefcase.dm [hunk 16]
```diff
@@ -20,10 +20,78 @@
...
+	else
+		return ..()
+
+/obj/item/storage/briefcase/false_bottomed/attack_hand(mob/user)
+	if(bottom_open && stored_item)
+		user.put_in_hands(stored_item)
+		to_chat(user, "You pull out \the [stored_item] from \the [src]'s false bottom.")
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/briefcase.dm]: 😱

**Pull request merged**